### DISCUSSION TITLE
Swift 6 warning: No `async` operations occur

### DISCUSSION THREAD

**Discussion by seanmrich** [pointfreeco/swift-dependencies#227] [Q&A] [answered]
I'm auditing an app for Swift 6 support, and I ran into warnings in a TCA dependency while using `ActorIsolated`. First, this code works as expected:
```swift
let isolated = ActorIsolated([Int: String]())
let s = await isolated.value[0]
```

So far, so good. Now I introduce a simple dependency and initialize it using an inline `ActorIsolated` value:
```swift
struct Client: Sendable {
  var key: @Sendable (Int) async throws -> String
}

let client: Client = {
  let isolated = ActorIsolated([Int: String]())
  return Client { key in
    await isolated.value[key] ?? ""
  }
}()
```

Again, no problem. But when I add it as a `DependencyKey`, the warning appears:
```swift
extension Client: DependencyKey {
  static let liveValue: Client = {
    let isolated = ActorIsolated([Int: String]())
    return Client { key in
      await isolated.value[key] ?? ""
   // 👆 No 'async' operations occur within 'await' expression
    }
  }()
}
```

I can't see anything in the declaration of `DependencyKey` that explains the difference. Is there something I missed? Perhaps a Swift bug?

**Comment by stephencelis** [comment, marked as answer]
@seanmrich This looks like a Swift bug. I can reproduce outside of swift-dependencies and mainly seems to be the static `let` assignment:

```swift
let f: @Sendable () async -> Int = {
  actor A { var p = 0 }
  return await A().p
}
```

Want to file a bug with Apple and ask what the expected behavior is?

In your case you can change to a computed `static var` to fix. But I'd also suggest not using `ActorIsolated`. We think that `LockIsolated` should almost always be preferred, and are having enough of a hard time motivating `ActorIsolated` that we will probably deprecate it soon.

**Reply by seanmrich** [reply]
Thanks for the help! I started a discussion over at the [swift forums](https://forums.swift.org/t/actor-access-in-static-let/72548) to be sure there isn't something new that explains the behavior. Using `LockIsolated` is good advice for my immediate issue, though. Thanks again!
